### PR TITLE
security and front-end

### PR DESCRIPTION
**Pull request by outsidereal**

### CODE REVIEW

File: src/main/java/com/epam/training/auction/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public String processRegistration(@RequestParam String login, @RequestParam String password, @RequestParam String confirmPassword) {
+        if (verifyPassword(password, confirmPassword)) {
+            long id = usersService.save(new User(login, password));
+            return "redirect:/user/" + id;
+        }
+        //// TODO: 11/20/2015 info about invalid data
+        return "register";
+    }
+
+    private boolean verifyPassword(String password, String confirmPassword) {
```
Comment (adbronowicki) [inline src/main/java/com/epam/training/auction/controllers/UserController.java:52]: Can we move it to src/main/webapp/js/registration-validator.js? We already have there password length checking.

Comment (outsidereal) [inline src/main/java/com/epam/training/auction/controllers/UserController.java:52]: Typically validation performs on both sides. Client-side and server-side.

File: src/main/java/com/epam/training/auction/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.epam.training.auction.service;
+
+import com.epam.training.auction.user.User;
+
+public interface UserService {
```
Comment (adbronowicki) [inline src/main/java/com/epam/training/auction/service/UserService.java]: Shouldn't we combine it with UsersService?

File: src/main/java/com/epam/training/auction/user/CustomUserDetails.java [hunk 3]
```diff
@@ -10,12 +10,12 @@
 
     private static final long serialVersionUID = -4261337858630172703L;
     
-    public CustomUserDetails(long id, String username, String password){
-        super(id, username, password);
+    public CustomUserDetails(String username, String password){
```
Comment (adbronowicki) [inline src/main/java/com/epam/training/auction/user/CustomUserDetails.java:13]: Do we really need this? We have also CustomUserDetails(User user) which calls the same super constructor.

**Pull request merged**